eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -1309,7 +1318,10 @@ registerPool = do
 
 -- | Performs the action without running the fix-up function on any transactions
 withNoFixup :: ImpTestM era a -> ImpTestM era a
-withNoFixup = local $ iteDoTxFixupL .~ False
+withNoFixup = local $ iteFixupL .~ pure
+
+withPostFixup :: (Tx era -> ImpTestM era (Tx era)) -> ImpTestM era a -> ImpTestM era a
```

Comment:
```suggestion
withFixup :: (Tx era -> ImpTestM era (Tx era)) -> ImpTestM era a -> ImpTestM era a
```

Reply:
Oh, sorry nvm, I thought it overwrites the fixup function, but it only composes an arrow with it. I think we should also just have `withFixup` that overwrites the function completely. That would probably solve one of the problems you mentioned.

Reply:
Yes! Alexey suggested the same, called `withCustomFixup`.  I added this, thanks.

Reply:
Ah actually `withCustomFixup` is more generic , allowing to compose the configured fixup (or nothing). 
I agree with you, I would like to also have a function that overrides the fixup entirely

Reply:
I'd rename `withCustomFixup` to `withModifyFixup`